src/hypothesis/extra/pandas/impl.py

```diff
@@ -253,28 +253,27 @@ def series(
             elements strategy varies, then so will the resulting dtype of the
             series.
 
-        min_size: the minimum number of entries in the resulting Series.
+        index: If not None, a strategy for generating indexes for the resulting
+            Series. This can generate either pandas.Index objects or any
```

Comment:
I'd actually require it to generate `pd.Index` objects - users can easily `.map(pd.Index)` on their sequence strategy, and that makes the weird magical stuff their problem instead of ours - Pandas is already so flexible that I think having a very tight API on our end will help a lot.

(and likewise for the dataframes strategy)